src/mp_integer.hpp

```diff
@@ -844,6 +865,54 @@ struct static_integer
 		}
 		clear_extra_bits();
 	}
+	// lshift by n bits.
+	int lshift(unsigned long n)
+	{
+		using size_type = typename limbs_type::size_type;
+		if (n == 0) {
```

Comment:
Here I'd prefer ``n == 0u``, as a general principle I prefer to avoid the risk of signed-unsigned conversions (here it should be ok, but yay consistency).